cms/articles/models.py

```diff
@@ -426,3 +436,35 @@ def serve_preview(self, request: "HttpRequest", mode_name: str) -> "TemplateResp
         if mode_name == "related_data":
             return cast("TemplateResponse", self.related_data(request))
         return cast("TemplateResponse", super().serve_preview(request, mode_name))
+
+    def get_serialized_corrections_and_notices(
+        self, request: "HttpRequest"
+    ) -> tuple[list[dict[str, Any]], list[dict[str, Any]]]:
+        """Returns a list of corrections and notices for the page."""
+        corrections = (
+            [
+                serialize_correction_or_notice(
+                    correction,
+                    superseded_url=self.get_url(request)
```

Comment:
suggestion: move `self.get_url(request)` outside of the loop. No need recalculate the same thing multiple times